addon/mixins/map-route-csw-loader.js

```diff
@@ -0,0 +1,92 @@
+/**
+  @module ember-flexberry-gis
+*/
+
+import Ember from 'ember';
+import {
+  Query
+} from 'ember-flexberry-data';
+
+/**
+  Mixin thats load data for use in add layers dialog by CSW settings.
+
+  @class MapRouteCswLoaderMixin
+  @extends <a href="http://emberjs.com/api/classes/Ember.Mixin.html">Ember.Mixin</a>
+*/
+export default Ember.Mixin.create({
+  /**
+  Name of CSW connection model projection to be used as record's properties limitation.
+
+  @property cswConnectionProjection
+  @type String
+  @default 'MapE'
+*/
+  cswConnectionProjection: 'CswConnectionE',
+
+  /**
+    Name of CSW connection model to be used as form's record type.
+
+    @property modelName
+    @type String
+    @default 'new-platform-flexberry-g-i-s-csw-connection'
+  */
+  cswConnectionModelName: 'new-platform-flexberry-g-i-s-csw-connection',
+
+  /**
+    Loaded CSW connections.
+
+    @property cswConnections
+    @type Object[]
+    @default null
+  */
+  cswConnections: null,
+
+  /**
+    [Model hook](http://emberjs.com/api/classes/Ember.Route.html#method_model) that returns a map project for current route.
+
+    @method model
+    @param {Object} params
+    @param {Object} transition
+    @return {<a href="http://emberjs.com/api/classes/RSVP.Promise.html">Ember.RSVP.Promise</a>}
+    Promise which will return map project related to current route
+  */
+  model(params) {
+    this._loadCswConnections(params);
```

Comment:
Is in normal that CSW-connections request & following request of super class executes separately in same time, but route will wait only for the second one ?